telegram-translation-shortcuts.user.js

```diff
@@ -70,16 +70,53 @@ function scrollItems (down) {
  * Clicks the 'Add Translation' button
  */
 function addTranslation () {
-  document.getElementsByClassName('key-add-suggestion-header').item(0).click()
+  // don't click if input box is expanded
+  var input_wrapper_collapsed = (document.getElementsByClassName('key-add-suggestion-wrap')[0].className == 'key-add-suggestion-wrap');
+  
+  if (!input_wrapper_collapsed)   // don't toggle 'add' and 'cancel'
+    document.getElementsByClassName('key-add-suggestion-header').item(0).click() 
+  else
+    document.getElementsByClassName('input form-control tr-form-control key-add-suggestion-field').item(0).focus() //fixme: throws undefined error
 }
 
 /**
  * Clicks the edit icon
  */
 function editTranslation () {
-  document.getElementsByClassName('ibtn key-suggestion-edit').item(0).click()
+  //TODO: don't click if input box is expanded
+  document.getElementsByClassName('ibtn key-suggestion-edit').item(0).click()  
 }
 
+/**
+ * Clicks the cancel button
+ */
+function cancelTranslation () {
+  // restrict when cancel button can be pressed
```

Comment:
Nitpick: golden rule of commenting code – comment why, not what you are doing. It’s pretty obvious you are restricting it.

(And an even smaller nitpick – sometimes you start comments with small letters, sometimes with big; unnecessary inconsistency imho.)